### PR TITLE
TS teams

### PR DESCRIPTION
**Pull request by cjb**

### CODE REVIEW

**Comment by cjb** [comment]
@keybase/react-hackers Ready for review!

File: shared/teams/team/header/index.tsx [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
+  openTeam: boolean
+  role: Types.MaybeTeamRoleType
+  showingMenu: boolean
+  teamname: Types.Teamname
+  onAddSelf: () => void
+  onChat: () => void
+  onEditDescription: () => void
+  onEditIcon: (image?: Response) => void
+  onFilePickerError: (error: Error) => void
+  onRename: () => void | null
```
Comment (buoyad) [inline shared/teams/team/header/index.tsx]: looks like the bug recently fixed in `flow-to-ts`. should be `(() => void) | null`

File: shared/teams/team/menu-container.tsx [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+// @ts-ignore not typed yet
+import {MenuItem} from '../../common-adapters/floating-menu/menu-layout'
 import {FloatingMenu} from '../../common-adapters'
 
 type OwnProps = {
-  attachTo: () => ?React.Component<any>,
-  onHidden: () => void,
-  teamname: string,
-  visible: boolean,
+  attachTo: () => React.Component<any> | null
```
Comment (buoyad) [inline shared/teams/team/menu-container.tsx]:
```suggestion
  attachTo: (() => React.Component<any>) | null
```

File: shared/teams/team/menu-container.tsx [hunk 3]
```diff
@@ -88,11 +88,12 @@ const mergeProps = (stateProps, dispatchProps, ownProps: OwnProps) => {
 }
 
 type Props = {
-  attachTo: () => ?React.Component<any>,
-  items: Array<MenuItem | 'Divider' | null>,
-  onHidden: () => void,
-  visible: boolean,
+  attachTo: () => React.Component<any> | null
```
Comment (buoyad) [inline shared/teams/team/menu-container.tsx]: same here

File: shared/teams/team/nav-header/index.tsx [hunk 4]
```diff
@@ -99,9 +101,9 @@ export const HeaderTitle = (props: HeaderTitleProps) => (
   </Kb.Box2>
 )
 
-type SubHeaderProps = {|
-  onAddSelf: ?() => void,
-|}
+type SubHeaderProps = {
+  onAddSelf: () => void | null
```
Comment (buoyad) [inline shared/teams/team/nav-header/index.tsx]: this and L62-64

File: shared/teams/team/settings-tab/index.tsx [hunk 5]
```diff
@@ -270,6 +266,7 @@ export class Settings extends React.Component<Props, State> {
   }
 
   setBoolSettings = (key: any) => (newSetting: boolean) => {
+    // @ts-ignore not sure how to type this
```
Comment (buoyad) [inline shared/teams/team/settings-tab/index.tsx]: Maybe you could split `State` into the bool properties and the rest, and type `key: keyof BoolState` here. No need to do it now though.

File: shared/teams/team/settings-tab/retention/index.tsx [hunk 6]
```diff
@@ -1,49 +1,47 @@
...
+type PropsWithOverlay<P> = {} & P & Kb.OverlayParentProps
 
 class _RetentionPicker extends React.Component<PropsWithOverlay<Props>, State> {
   state = {
     items: [],
     saving: false,
     selected: retentionPolicies.policyRetain,
   }
-  _timeoutID: TimeoutID
+  _timeoutID: number
```
Comment (buoyad) [inline shared/teams/team/settings-tab/retention/index.tsx]: `NodeJS.Timeout`

File: shared/teams/team/settings-tab/retention/warning/container.tsx [hunk 7]
```diff
@@ -23,19 +28,19 @@ const mapDispatchToProps = (dispatch, ownProps) => {
...
     onBack: () => {
       dispatch(RouteTreeGen.createNavigateUp())
-      const onCancel: ?() => void = Container.getRouteProps(ownProps, 'onCancel')
+      const onCancel: () => void | null = Container.getRouteProps(ownProps, 'onCancel')
       onCancel && onCancel()
     },
     onConfirm: () => {
       dispatch(RouteTreeGen.createNavigateUp())
-      const cb: ?() => void = Container.getRouteProps(ownProps, 'onConfirm')
+      const cb: () => void | null = Container.getRouteProps(ownProps, 'onConfirm')
```
Comment (buoyad) [inline shared/teams/team/settings-tab/retention/warning/container.tsx]: Parens around function here and L31

**Pull request merged**